src/services/createRouter.ts

```diff
@@ -187,8 +202,8 @@ export function createRouter<const TRoutes extends Routes, const TOptions extend
     history.startListening()
   }
 
-  const resolve: RouterResolve<TRoutes> = (
-    source: RoutesName<TRoutes>,
+  const resolve: RouterResolve<TRoutes | TPlugin['routes']> = (
```

Comment:
I think I'd be tempted to define a type for this as like the first line of createRouter implementation

```ts
type PossibleRoutes = TRoutes | TPlugin['routes']
```

Reply:
Yeah tempting but I think this reads nicely still. If it gets more complex I'm down to alias it.